exporter/azureblobexporter/factory.go

```diff
@@ -0,0 +1,63 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azureblobexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azureblobexporter"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr             = "azureblob"
+	logsContainerName   = "logs"
+	tracesContainerName = "traces"
+)
+
+// NewFactory returns a factory for Azure Blob exporter.
+func NewFactory() component.ExporterFactory {
+	return component.NewExporterFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithTracesExporter(createTracesExporter, component.StabilityLevelInDevelopment),
+		component.WithLogsExporter(createLogsExporter, component.StabilityLevelInDevelopment))
+}
+
+func createDefaultConfig() config.Exporter {
+	return &Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		Logs:             LogsConfig{ContainerName: logsContainerName},
+		Traces:           TracesConfig{ContainerName: tracesContainerName},
+	}
+}
+
+func createTracesExporter(
+	ctx context.Context,
+	set component.ExporterCreateSettings,
+	cfg config.Exporter,
+) (component.TracesExporter, error) {
+	return exporterhelper.NewTracesExporter(context.TODO(), set, cfg, onTraceData)
```

Comment:
You can pass the ctx argument instead of `context.TODO()` here.
```suggestion
	return exporterhelper.NewTracesExporter(ctx, set, cfg, onTraceData)
```